core/core-keybindings.el

```diff
@@ -46,50 +50,40 @@ used as the prefix command."
          (major-mode-prefix-emacs (concat dotspacemacs-major-mode-emacs-leader-key " " (substring prefix 1))))
     (unless long-name (setq long-name name))
     (let ((prefix-name (cons name long-name)))
-      (if (fboundp 'which-key-declare-prefixes-for-mode)
-          (progn
-            (which-key-declare-prefixes-for-mode mode
-              full-prefix-emacs prefix-name
-              full-prefix prefix-name)
-            (when (and is-major-mode-prefix dotspacemacs-major-mode-leader-key)
-              (which-key-declare-prefixes-for-mode mode major-mode-prefix prefix-name))
-            (when (and is-major-mode-prefix dotspacemacs-major-mode-emacs-leader-key)
-              (which-key-declare-prefixes-for-mode mode major-mode-prefix-emacs prefix-name)))
-        (define-prefix-command command)
-        (evil-leader/set-key-for-mode mode prefix command)))))
+      (which-key-declare-prefixes-for-mode mode
+        full-prefix-emacs prefix-name
+        full-prefix prefix-name)
+      (when (and is-major-mode-prefix dotspacemacs-major-mode-leader-key)
+        (which-key-declare-prefixes-for-mode mode major-mode-prefix prefix-name))
+      (when (and is-major-mode-prefix dotspacemacs-major-mode-emacs-leader-key)
+        (which-key-declare-prefixes-for-mode mode major-mode-prefix-emacs prefix-name)))))
+
+(defun spacemacs/set-keys (key def &rest bindings)
+  (while key
+    (define-key spacemacs-default-map (kbd key) def)
+    (setq key (pop bindings) def (pop bindings))))
+
+(defun spacemacs//init-major-mode-map (mode map)
+  (let ((prefix (intern (format "%s-prefix" map))))
+    (or (boundp prefix)
+        (progn
+          (eval
+           `(bind-map ,map
+              :prefix-cmd ,prefix
+              :major-modes (,mode)
+              :keys (,dotspacemacs-major-mode-emacs-leader-key
+                     ,(concat dotspacemacs-emacs-leader-key " m"))
+              :evil-keys (,dotspacemacs-major-mode-leader-key
+                          ,(concat dotspacemacs-leader-key " m"))))
+          (boundp prefix)))))
+
+(defun spacemacs/set-keys-for-major-mode (mode key def &rest bindings)
```

Comment:
Same here, also don't be shy with details because those functions will be at the heart of the distro.